### PR TITLE
Prevent screen reader from reading modal twice

### PR DESCRIPTION
**Pull request by ataker**
## Chromatic

https://prevent-double-screen-reader-modal--60f9b557105290003b387cd5.chromatic.com

## Description
Closes [Screen readers are reading modal content out twice #1849](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1849)

## Testing done
Tested with Mac Voiceover in chrome and firefox

## Acceptance criteria
- [ ] Screen reader doesn't repeat twice when opening a modal.

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by ataker** [comment]
I can't get this to consistently work with the proper accessibility tags with the `alertdialog` role. Removing it and all modals will be considered `dialog`.

File: packages/web-components/src/components/va-modal/va-modal.tsx [hunk 1]
```diff
@@ -443,7 +437,7 @@ export class VaModal {
                       <li class="usa-button-group__item">
                         {!unstyled && (
                           <va-button
-                            onClick={e => this.handlePrimaryButtonClick(e)}
+                            onClick={e => this.handleSecondaryButtonClick(e)}
```
Comment (ataker) [inline packages/web-components/src/components/va-modal/va-modal.tsx:440]: Not related to the issue, but fixing bug in USWDS modal where both secondary button was performing the action of the primary button.

**Review by Andrew565 — APPROVED** [review]
LGTM, as long as it's actually working. Didn't test with a screenreader.

File: packages/web-components/src/components/va-modal/va-modal.tsx [hunk 2]
```diff
@@ -475,8 +469,13 @@ export class VaModal {
       const bodyClass = status ? 'va-modal-alert-body' : 'va-modal-body';
       const titleClass = status ? 'va-modal-alert-title' : 'va-modal-title';
       return (
-        <Host aria-label={ariaLabel} aria-modal="true" role={ariaRole(status)}>
-          <div class={wrapperClass} tabIndex={-1}>
+        <Host>
+          <div class={wrapperClass}
+            role='dialog'
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-modal/va-modal.tsx:474]: If we are not toggling between "dialog" and "alertdialog" any longer, is there a utility method we can remove as well?

Comment (jamigibbs) [reply packages/web-components/src/components/va-modal/va-modal.tsx:474]: Oh my bad! It is removed. :D

**Comment by ataker** [comment]
Feedback I've gotten is that they would prefer the `alertdialog` role with the chance of double reading rather than have everything be `dialog`, so reverting that. This isn't completely fixed, but it does seem isolated to Mac Voiceover and never happens on the first open, so hopefully this will be a rare incident.

**Pull request merged**